// @ionic-native/speech-recognition/__ivy_ngcc__/ngx/index.js
import { __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import { Observable } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
var SpeechRecognition = /** @class */ (function (_super) {
    __extends(SpeechRecognition, _super);
    function SpeechRecognition() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    SpeechRecognition.prototype.isRecognitionAvailable = function () { return cordova(this, "isRecognitionAvailable", {}, arguments); };
    SpeechRecognition.prototype.startListening = function (options) { return cordova(this, "startListening", { "callbackOrder": "reverse", "observable": true }, arguments); };
    SpeechRecognition.prototype.stopListening = function () { return cordova(this, "stopListening", {}, arguments); };
    SpeechRecognition.prototype.getSupportedLanguages = function () { return cordova(this, "getSupportedLanguages", {}, arguments); };
    SpeechRecognition.prototype.hasPermission = function () { return cordova(this, "hasPermission", {}, arguments); };
    SpeechRecognition.prototype.requestPermission = function () { return cordova(this, "requestPermission", {}, arguments); };
    SpeechRecognition.pluginName = "SpeechRecognition";
    SpeechRecognition.plugin = "cordova-plugin-speechrecognition";
    SpeechRecognition.pluginRef = "plugins.speechRecognition";
    SpeechRecognition.repo = "https://github.com/pbakondy/cordova-plugin-speechrecognition";
    SpeechRecognition.platforms = ["Android", "iOS"];
SpeechRecognition.ɵfac = /*@__PURE__*/ function () { var ɵSpeechRecognition_BaseFactory; return function SpeechRecognition_Factory(t) { return (ɵSpeechRecognition_BaseFactory || (ɵSpeechRecognition_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(SpeechRecognition)))(t || SpeechRecognition); }; }();
SpeechRecognition.ɵprov = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjectable({ token: SpeechRecognition, factory: function (t) { return SpeechRecognition.ɵfac(t); } });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpeechRecognition, [{
        type: Injectable
    }], null, null); })();
    return SpeechRecognition;
}(IonicNativePlugin));
export { SpeechRecognition };
